### PR TITLE
Added reuse able workflow

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Replace workflows with re use able workflow

Flytetools: https://github.com/flyteorg/flytetools/pull/43/files

Test: 
Release : https://github.com/evalsocket/flyteadmin/actions/runs/1788340280
Pull Request: https://github.com/flyteorg/flyteadmin/actions/runs/1788397103
## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: .github/workflows/upgrade_automtion.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Upgrade Automation
```
Comment (EngHabu) [inline .github/workflows/upgrade_automtion.yml:1]: Do you want to keep this file named the same or will you follow up with an update to this? https://github.com/flyteorg/boilerplate/blob/master/.github/workflows/master.yaml#L19

**Review by EngHabu — DISMISSED** [review]
Awesome! much cleaner

File: .github/workflows/upgrade_automtion.yml (hunk 1 above)
Comment (evalsocket) [reply .github/workflows/upgrade_automtion.yml:1]: Now we have only one workflow for upgrade all component like flyteidl, boilerplate etc

**Review by EngHabu — APPROVED** [review]
